packages/muelu/example/advanced/levelwrap/LevelWrap.cpp

```diff
@@ -117,6 +118,13 @@ namespace MueLuExamples {
     belosProblem->setRightPrec(belosPrec);
     belosProblem->setProblem(X,B);
     Belos::SolverFactory<Scalar, MV, OP> BelosFactory;
+
+    // We need to register our manager - or do we want these auto registered?
```

Comment:
Why can't the Belos::SolverFactory method call the registration function?  If the function is idempotent and low-cost to call, then we could just call it each time users try to create a solver.

Reply:
Please see general note below on this call - this can be automatic but then i think we'll have to put the includes back in the SolverFactory header for all the managers.